UOP1_Project/Assets/Scripts/Characters/Character.cs

```diff
@@ -72,6 +88,15 @@ private void Update()
             //Full speed ground movement
             movementVector = inputVector * speed;
 
+            if ((Math.Abs(movementVector.x) > 0.01f || Math.Abs(movementVector.z) > 0.01f) && characterController.isGrounded)
```

Comment:
You can set these conditions as variables to reduce some duplication and make it more readable.  Also, what is the significance of 0.01f, rather than say 0.001f? ( New to Unity / Maybe that's a Unity thing? )

Reply:
This is probability to deal with analogical deadzones. Since they are using the new input system and there is a property for deadzones, this `0.01f` could be replaced for 0F

Reply:
Also, you could rewrite this part as follow:
```suggestion
            bool isMoving = Math.Abs(inputVector.sqrMagnitude) > 0F;
            if (isMoving && characterController.isGrounded)
```

Reply:
Writing it like that could introduce floating point error bugs. The reason for checking if the absolute value is greater than 0.01 is to avoid a floating point error from technically making the number greater than 0 even though in practical terms it is 0. It would be better to either define a threshold (which that 0.01f would be) or to use `float.Epsilon` which is a very small value that's specifically used for checks like this, here's a [StackOverflow post](https://stackoverflow.com/questions/30216575/why-float-epsilon-and-not-zero) that explains it better

Reply:
I used `float.Epsilon` instead of 0.01f to avoid using magic numbers, and created a separate variable to check if the player is moving or not, instead of having the whole comparison on the if statement